### PR TITLE
Incorporate tbdex vector results into report

### LINKED ISSUE #83: Add tbdex vector results to generated report

### PR DESCRIPTION
**Pull request by phoebe-lew**
- Add hermit
- Readme and some report gen changes
- Handle tbdex test results

closes #83

### CODE REVIEW

**Comment by mistermoe** [comment]
@phoebe-lew would it make more sense to init hermit inside of the `reports` dir vs at the root?

**Comment by phoebe-lew** [comment]
@mistermoe hmm could do either, yeah! I tended to open the entire repo in Goland, but not sure what others usually do.
@finn-tbd @nitro-neal do you usually open just the generator, or the entire repo?

@alecthomas will it be fine to have the `bin` dir nested inside `reports`?

**Comment by alecthomas** [comment]
Definitely preferable at the root, so that it activates automatically when you cd into it.

**Comment by finn-tbd** [comment]
I usually open the root. I found thought it was a little confusing to move the go.mod into a subdir but that was what was requested 🤷

File: reports/sdks.go [hunk 1]
```diff
@@ -28,6 +29,25 @@ var (
 			FeatureRegex: regexp.MustCompile(`web5\.sdk\.\w+.Web5TestVectors(\w+)`),
 			VectorRegex:  regexp.MustCompile(`(\w+)\(\)`),
 			VectorPath:   "test-vectors",
+			Type:         "web5",
+		},
+		{
+			Name:         "tbdex-js",
+			Repo:         "TBD54566975/tbdex-js",
+			ArtifactName: "junit-results",
+			FeatureRegex: regexp.MustCompile(`TbdexTestVectors(\w+)`),
```
Comment (nitro-neal) [inline reports/sdks.go:38]: so you have been naming your tests:
TbdexTestVectorsClassFeature ?

for example: 
`TbdexTestVectorsRfqCreate` ?

**Review by nitro-neal — APPROVED** [review]

File: reports/sdks.go (hunk 1 above)
Comment (phoebe-lew) [reply reports/sdks.go:38]: Ah yep, Think I have a copy paste error for the vector regex though!

File: reports/html.go [hunk 2]
```diff
@@ -17,34 +17,51 @@ func sanatizeHTML(dirty error) string {
...
 	slog.Info("writing html report", "reports", len(reports))
 
 	testmap := make(map[string]map[string]bool)
+	tbdexTestMap := make(map[string]map[string]bool)
 	for _, report := range reports {
 		for category, tests := range report.Results {
 			if _, ok := tests[category]; !ok {
-				testmap[category] = map[string]bool{}
+				if report.SDK.Type == "web5" {
+					testmap[category] = map[string]bool{}
```
Comment (jiyoontbd) [inline reports/html.go:34]: nit: could we rename this to `web5TestMap`?

Comment (decentralgabe) [reply reports/html.go:34]: you shouldn't need to initialize like this..., but if it breaks pls use `make(map[string]bool)`

File: reports/html.go [hunk 3]
```diff
@@ -17,34 +17,51 @@ func sanatizeHTML(dirty error) string {
...
-	Tests   map[string][]string
+	Reports    []Report
+	Web5Tests  map[string][]string
+	TbDEXTests map[string][]string
 }
 
 func WriteHTML(reports []Report, filename string) error {
 	slog.Info("writing html report", "reports", len(reports))
 
 	testmap := make(map[string]map[string]bool)
```
Comment (decentralgabe) [inline reports/html.go:28]:
```suggestion
	testMap := make(map[string]map[string]bool)
```

for consistency

**Review by decentralgabe — APPROVED** [review]
excited to try out hermit!

File: bin/go [hunk 4]
```diff
@@ -0,0 +1 @@
+.go-1.21.5.pkg
```
Comment (decentralgabe) [inline bin/go:1]: can be 1.21.6 now?

**Pull request merged**